### ISSUE TITLE
Green artefacts

### ISSUE THREAD

**Issue by pharshdev** [befovy/fijkplayer#201]
Green artefacts when playing videos on Firestick 4k.
<img width="380" alt="Screenshot 2020-03-23 at 7 59 20 AM" src="https://user-images.githubusercontent.com/15626305/77274710-8a78d100-6cdc-11ea-8def-7a8ca1d98007.png"> 

Audio and video works fine, its just the green thing!

**Comment by befovy** [comment]
Can you provide the log?

**Comment by befovy** [comment]
Have you tried hard video decoding and soft video decoding?
Have this issue on other devices?

**Comment by pharshdev** [comment]
> Have you tried hard video decoding and soft video decoding?

No, I've used the example as is. I'm not sure how to use the flags if that's what you meant. Please tell.

> Have this issue on other devices?

So far I have tested on firestick 4k and non 4k, android tv, shield and all of them on both 4k and 1080 displays. Only firestick 4k gave the green issue on both 4k and 1080 displays. 

Now you could blame the hardware but then Kodi which uses the same ffplay, plays the video just without any issues.

**Comment by befovy** [comment]
Enable mediacodec using setOption.

```
player.setOption(FijkOption.playerCategory, "mediacodec-all-videos" , 1);
```

See https://github.com/befovy/fijkplayer/issues/176#issuecomment-587260649 for details

**Comment by pharshdev** [comment]
I have already set all the flags mentioned in the comment [#176](https://github.com/befovy/fijkplayer/issues/176#issuecomment-587260649). Doesn't solve the green issue unfortunately.

**Comment by befovy** [comment]
@pharshdev  Have you tried the native ijkplayer demo?
https://github.com/befovy/ijkplayer

Or have you tried other flutter video player plugin?

**Comment by pharshdev** [comment]
> @pharshdev Have you tried the native ijkplayer demo?
> https://github.com/befovy/ijkplayer

Yes, the issue persists with native ijkplayer too. 

> Or have you tried other flutter video player plugin?

VLC player and flt player works fine after playing with their options.

**Comment by befovy** [comment]
@pharshdev Thansk for the reply.  But I can't give a solution for this problem

**Comment by pharshdev** [comment]
Oh my god, I'm so hyped right now! Not sure how but I actually found a way around the green issue!

Looks like if i comment out mediacodec option, i do not get the green issue. But then the videoplayback is choppy. If i uncomment it, the playback is smooth but there is this green artefact!>

```
@override
  void initState() {
    super.initState();
    player.setDataSource(widget.link, autoPlay: true);
    player.setOption(FijkOption.playerCategory, "reconnect", 5);
    player.setOption(FijkOption.playerCategory, "framedrop", 5);
    player.setOption(FijkOption.playerCategory, "enable-accurate-seek", 1);
   // player.setOption(FijkOption.playerCategory, "mediacodec_all_videos", 1);
    player.setOption(FijkOption.playerCategory, "packet-buffering", 1);
    player.setOption(FijkOption.playerCategory, "soundtouch", 1);
    player.setOption(FijkOption.playerCategory, "start-on-prepared", 1);
    player.setOption(FijkOption.formatCategory, "fflags", "fastseek");
  }
```
I have been doing some reading and it looks like it has something to do with YUV format. Idk can you please take a look now [https://github.com/befovy/fijkplayer/issues/176](url). Thank you so much, this player otherwise is excellent!
